### ISSUE TITLE
Entry with IsPassword=true exposes entered text to screen readers (Windows Narrator)

### ISSUE THREAD

**Issue by Muthukumaran-G** [dotnet/maui#33577]
### Description

**Bug Description**
When an Entry (or Editor) is configured as a password field (IsPassword=true), screen readers announce the characters that the user types, effectively exposing the password. This violates expected password masking behavior and constitutes a privacy/security concern.

`<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"`
            ` xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"`
             `x:Class="PasswordNarratorIssue.MainPage"`
             `Title="Password Narrator Issue">`
    `<VerticalStackLayout>`
        `<Label Text="Password" />`
        `<Entry x:Name="PasswordEntry"`
              ` IsPassword="True"`
               `Placeholder="Enter password" />`
        `<Button Text="Login" />`
    `</VerticalStackLayout>`
`</ContentPage>`

### Steps to Reproduce

1. Create a new .NET MAUI app (or use the sample below).
2. Add an Entry with IsPassword="True".
3. Open the narrator
4. Enter the password in entry.

Actual behavior: Note that the entered characters are prompted out by narrator.
Expected behavior: Narrator should say either hidden or Dot

### Version with bug

10.0.30

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10 SDK 10.0.22000

### Did you find any workaround?

No

### Relevant log output

```shell

```

**Comment by Vignesh-SF3580** [comment]
This issue has been validated in Visual Studio Professional 2026 using MAUI versions 9.0.120, 10.0.0, and 10.0.30, and it is reproducible on the Windows platform using Narrator.
